tools/DatasetExtraction/dataset.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABCMeta, abstractmethod
+from accessify import protected
+from chord import Chord, Song
+import numpy as np
+from glob import glob
+
+class Dataset:
+    #i - song, j - chord
+    songlist = [[]]
+
+
+    @protected
+    def validate(self):
+        pass
+
+    @protected
+    def load(self):
+        pass
+
+    def get_hmm_matrix_i(self):
+        pass
+
+    def get_hmm_matrix_t(self):
+        pass
+
+    def get_hmm_matrix_e(self):
+        pass
+
+
+class Iso_Dataset(Dataset):
+
+    # assume we have chord_id dictionary which maps Chords into ids
+    def get_hmm_matrix_i(self, chord_id, dir_path):

```

Comment:
What's the point of having this function duplicated in each child class? We discussed earlier that this is a parent class job.